packages/ui/src/component/UITransform.ts

```diff
@@ -35,15 +65,348 @@ export class UITransform extends Transform {
+    this._top = value;
+    if (this._verticalAlignment & VerticalAlignmentFlags.Top) {
+      this._onPositionChanged();
+      this._verticalAlignment & VerticalAlignmentFlags.Bottom && this._onSizeChanged();
+    }
+  }
+
+  /**
+   * Bottom margin used in vertical alignment formulas.
+   */
+  get bottom(): number {
+    return this._bottom;
+  }
+
+  set bottom(value: number) {
+    if (!Number.isFinite(value)) return;
+    if (MathUtil.equals(value, this._bottom)) return;
+    this._bottom = value;
+    if (this._verticalAlignment & VerticalAlignmentFlags.Bottom) {
+      this._onPositionChanged();
+      this._verticalAlignment & VerticalAlignmentFlags.Top && this._onSizeChanged();
+    }
+  }
+
+  /**
+   * Vertical middle offset relative to parent's middle.
+   */
+  get middle(): number {
+    return this._middle;
+  }
+
+  set middle(value: number) {
+    if (!Number.isFinite(value)) return;
+    if (MathUtil.equals(value, this._middle)) return;
+    this._middle = value;
+    this._verticalAlignment & VerticalAlignmentFlags.Middle && this._onPositionChanged();
+  }
+
+  /**
+   * Local position.
+   */
+  override get position(): Vector3 {
+    if (this._isContainDirtyFlag(UITransformModifyFlags.LocalPosition)) {
+      const parentRect = (this._getParentTransform() as unknown as UITransform)?._getRect?.();
+      if (!!parentRect) {
+        const position = this._position;
+        // @ts-ignore
+        position._onValueChanged = null;
+        const rect = this._getRect();
+        switch (this.horizontalAlignment) {
+          case HorizontalAlignmentFlags.Left:
+          case HorizontalAlignmentFlags.LeftAndRight:
+            position.x = parentRect.x - rect.x + this._left;
+            break;
+          case HorizontalAlignmentFlags.Center:
+            position.x = parentRect.x + parentRect.width * 0.5 - rect.x - rect.width * 0.5 + this._center;
+            break;
+          case HorizontalAlignmentFlags.Right:
+            position.x = parentRect.x + parentRect.width - rect.x - rect.width - this._right;
+            break;
+          default:
+            break;
+        }
+        switch (this.verticalAlignment) {
+          case VerticalAlignmentFlags.Top:
+            position.y = parentRect.y + parentRect.height - rect.y - rect.height - this._top;
+            break;
+          case VerticalAlignmentFlags.Middle:
+            position.y = parentRect.y + parentRect.height * 0.5 - rect.y - rect.height * 0.5 + this._middle;
+            break;
+          case VerticalAlignmentFlags.Bottom:
+          case VerticalAlignmentFlags.TopAndBottom:
+            position.y = parentRect.y - rect.y + this._bottom;
+            break;
+          default:
+            break;
+        }
+        // @ts-ignore
+        position._onValueChanged = this._onPositionChanged;
+      }
+      this._setDirtyFlagFalse(UITransformModifyFlags.LocalPosition);
+    }
+    return this._position;
+  }
+
+  override set position(value: Vector3) {
```

Comment:
It is not recommended to override Property inside the engine, because you have to override `get` and `set` at the same time.